tests/test_filling_fields.py

```diff
@@ -275,6 +275,20 @@ def test_filling_content_type_field(self):
         assert isinstance(dummy.content_type, ContentType)
         assert dummy.content_type.model_class() is not None
 
+    def test_iteratively_filling_generic_foreign_key_field(self):
+        """
+        Ensures private_fields are included in Baker.get_fields(), otherwise
+        calling next() when a GFK is in iterator_attrs would be bypassed.
+        """
+        objects = baker.make(models.Profile, _quantity=2)
+        dummies = baker.make(
+            models.DummyGenericForeignKeyModel,
+            content_object=iter(objects),
```

Comment:
What happens when no value is provided for this field? Consider adding a test for it.

Also, what happens to the `content_type` and `object_id` fields? Maybe there should be a test for those as well?

Reply:
This test might be related to the latter: https://github.com/model-bakers/model_bakery/blob/main/tests/test_filling_fields.py#L272.

Reply:
I've updated the test to also verify the `content_type` and `object_id` field contain the expected values. 

I'm unclear on whether the question "What happens when no value is provided for this field?" means what happens for the case `baker.make(models.DummyGenericForeignKeyModel)` or the case `baker.make(models.DummyGenericForeignKeyModel, content_object=iter([]))`, since this PR specifically fixes `iter` being broken on a GFK. Did you want a test for the latter? The former case is already present in an existing test.

Reply:
No, I think you are right: the case I was thinking of is already covered. No action needed.